core/controllers/certificate_assessment.py

```
@@ -118,48 +153,122 @@ class CertificateAssessmentOfferingByIdHandler(
+                topic_ids=topic_ids,
+                total_questions=total_questions,
+                time_limit_in_minutes=time_limit_in_minutes,
+                demonstrates=demonstrates,
+                async_status=self.normalized_payload['async_status'],
+            )
+        except (
+            certificate_assessment_services.CertificateAssessmentOfferingNotFoundException
+        ) as e:
+            raise self.NotFoundException(str(e)) from e
+        except utils.ValidationError as e:
+            raise self.InvalidInputException(e)
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)